### DISCUSSION TITLE
How to find the repository a given registered algorithm is associated with ?

### DISCUSSION THREAD

**Discussion by emileten** [MAAP-Project/Community#838] [Platform] [answered]
How can we find the repository a given registered algorithm is associated with ? I've tried [the `describeAlgorithm` method](https://github.com/MAAP-Project/maap-py/blob/d0f34e3440bf1121882892a5fe0b20484b413d43/maap/maap.py#L191) of the maap python client, it doesn't provide that information. Searching in the MAAP github project or slack didn't help. In this case, I wanted to understand how the (already registered) `gdal_hello_world` algorithm works.

**Comment by wildintellect** [comment]
1. An algorithm catalog is on the future features list.
2. The one in question is https://repo.maap-project.org/sshah/gdal-hello-world
3. @sujen1412 all the algorithm configs are stored in a single Gitlab repo now? Should we surface that location?

**Comment by rtapella** [comment]
I would suggest that “no” we don’t expose the Gitlab repo. We should have an endpoint to ask for the algorithms metadata though.

There is also a plan for a “algorithm catalog lite” to go with the registration UI. It should show the algorithm’s metadata also.

—
Rob Tapella

________________________________
From: Alex I. Mandel ***@***.***>
Sent: Friday, October 20, 2023 8:39:52 AM
To: MAAP-Project/Community ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXTERNAL] Re: [MAAP-Project/Community] How to find the repository a given registered algorithm is associated with ? (Discussion #838)


  1.  An algorithm catalog is on the future features list.
  2.  The one in question is https://repo.maap-project.org/root/gdal-hello-world<https://urldefense.us/v3/__https://repo.maap-project.org/root/gdal-hello-world__;!!PvBDto6Hs4WbVuu7!PxAsgSq6I53wBc2sQcTWuwe-9kAlaV57CieUHHOd1aHs0U0LMKLLcGzZsvJETEx-2HEUoFCrB0MOniPY9wM7iSoqD8L74oTwunQ$>
  3.  @sujen1412<https://urldefense.us/v3/__https://github.com/sujen1412__;!!PvBDto6Hs4WbVuu7!PxAsgSq6I53wBc2sQcTWuwe-9kAlaV57CieUHHOd1aHs0U0LMKLLcGzZsvJETEx-2HEUoFCrB0MOniPY9wM7iSoqD8L7PTzO4ZQ$> all the algorithm configs are stored in a single Gitlab repo now? Should we surface that location?

**Comment by emileten** [comment]
> The one in question is https://repo.maap-project.org/sshah/gdal-hello-world

Thank you @wildintellect !

> all the algorithm configs are stored in a single Gitlab repo now? 

That's a good question. I wasn't aware of this. For example, for the dps unit test, I was assuming this is the source of truth https://github.com/MAAP-Project/dps-unit-test, but based on what you're saying, this could be the one ? https://repo.maap-project.org/bsatoriu/dps-unit-test.

At least from a support standpoint I think it's important to be able to easily find a repo associated with a given algo. 

> I would suggest that “no” we don’t expose the Gitlab repo.

@rtapella curious, why do you think this isn't a good idea ? 

> We should have an endpoint to ask for the algorithms metadata though.

There is, but it doesn't show the github repo url.

> An algorithm catalog is on the future features list.

👍

**Reply by rtapella** [reply, marked as answer]
Having the Gitlab repo as a store for configurations is our current/new implementation of a metadata storage for algorithms. It may change in the future, especially as several people are now talking about making an Algorithm Catalog that's a bit more fully baked. Having an API wrapper that can abstract out however we're implementing it will mean less change for end-users and end-applications that are consuming this info, plus it will let us put appropriate authorization around the info as we improve that aspect.

**Reply by sujen1412** [reply]
Regarding exposing the location of the algorithm code, I am in favor of the metadata including the repo url and ref of the exact code that is being run in the job. 

This would allow the user to go to the url to dig into the code or open issue tickets with the developers.

**Reply by rtapella** [reply]
Oh, yea agree. The code repo and commit should be part of the metadata you get back from the API.
And that can/should be used by any GUI we write on top as well.

**Reply by emileten** [reply]
> I am in favor of the metadata including the repo url

Created an issue for this, as currently the API _does not_ return that information.

**Reply by rtapella** [reply]
Thank you Emile
